packages/engine/src/transform/systems/TransformSystem.ts

```diff
@@ -302,8 +303,10 @@ export default async function TransformSystem(world: World) {
         world.dirtyTransforms[entity] ||
         world.dirtyTransforms[getOptionalComponent(entity, LocalTransformComponent)?.parentEntity ?? 0] ||
         world.dirtyTransforms[getOptionalComponent(entity, ComputedTransformComponent)?.referenceEntity ?? 0] ||
-        hasComponent(entity, ComputedTransformComponent)
-      if (makeDirty) world.dirtyTransforms[entity] = true
+        hasComponent(entity, ComputedTransformComponent) ||
+        (hasComponent(entity, AnimationComponent) && getComponent(entity, AnimationComponent).animations.length > 0)
```

Comment:
For performance reasons, we don't want to add if statements to the transform system if possible. This can be done in another system, say the AnimationSystem, where we can set dirty upon updating the animation mixer.

Reply:
Yep that makes way more sense